src/usd/sdf_usd_parser/material.cc

```diff
@@ -0,0 +1,355 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "sdf_usd_parser/material.hh"
+
+#include <map>
+
+#include <ignition/math/Color.hh>
+#include <pxr/base/tf/stringUtils.h>
+#include <pxr/usd/sdf/types.h>
+#include <pxr/usd/usd/prim.h>
+#include <pxr/usd/usdShade/connectableAPI.h>
+#include <pxr/usd/usdShade/materialBindingAPI.h>
+
+#include "sdf/Pbr.hh"
+
+namespace usd
+{
+  void RemoveProperty(
+    pxr::UsdStageRefPtr &_stage,
+    pxr::SdfPath prim_path,
+    pxr::TfToken property_name)
+  {
+    for (auto &layer : _stage->GetLayerStack())
+    {
+      auto primSpec = layer->GetPrimAtPath(prim_path);
+      if (primSpec)
+      {
+        auto propertySpec =
+          layer->GetPropertyAtPath(
+            prim_path.AppendProperty(property_name));
+        if (propertySpec)
+        {
+          // primSpec.RemoveProperty(propertySpec);
+        }
+      }
+    }
+  }
+
+  template<typename T>
+  void CreateMaterialInput(
+    pxr::UsdPrim &_prim, std::string name, pxr::SdfValueTypeName vType, T value,
+    std::map<pxr::TfToken, pxr::VtValue> &_customData, pxr::TfToken displayName = pxr::TfToken(""),
+    pxr::TfToken displayGroup = pxr::TfToken(""), std::string doc = "",
+    pxr::TfToken colorSpace = pxr::TfToken(""))
+  {
+    auto shader = pxr::UsdShadeShader(_prim);
+    if (shader)
+    {
+      auto existingInput = shader.GetInput(pxr::TfToken(name));
+      pxr::SdfValueTypeName vTypeName;
+      if (vType.IsScalar())
+      {
+        vTypeName = vType.GetScalarType();
+      } else if (vType.IsArray())
+      {
+        vTypeName = vType.GetArrayType();
+      }
+      if (existingInput &&
+          existingInput.GetTypeName() != vTypeName)
+      {
+        // RemoveProperty(shader.GetPrim().GetPath(), existingInput.GetFullName());
+      }
+      auto surfaceInput = shader.CreateInput(
+        pxr::TfToken(name), vTypeName);
+      surfaceInput.Set(value);
+      auto attr = surfaceInput.GetAttr();
+
+      for (auto& [key, customValue]: _customData)
+      {
+        attr.SetCustomDataByKey(key, customValue);//pxr::TfToken("default"), pxr::VtValue(pxr::GfVec3f(0.2, 0.2, 0.2)));
+      }
+      if (!displayName.GetString().empty())
+      {
+        attr.SetDisplayName(displayName);
+      }
+      if (!displayGroup.GetString().empty())
+      {
+        attr.SetDisplayGroup(displayGroup);
+      }
+      if (!doc.empty())
+      {
+        attr.SetDocumentation(doc);
+      }
+      if (!colorSpace.GetString().empty())
+      {
+        attr.SetColorSpace(colorSpace);
+      }
+    }
+  }
+
+  pxr::UsdShadeMaterial ParseSdfMaterial(const sdf::Material *_material, pxr::UsdStageRefPtr &_stage)
+  {
+    auto looksPrim = _stage->GetPrimAtPath(pxr::SdfPath("/Looks"));
+    if (!looksPrim)
+    {
+      looksPrim = _stage->DefinePrim(pxr::SdfPath("/Looks"), pxr::TfToken("Scope"));
+    }
+
+    static int i = 0;
```

Comment:
Can you add a comment that describes what the variable `i` represents here, and/or change the name of the variable to something more descriptive?